distributed/protocol.py

```diff
@@ -64,35 +62,77 @@
     default_compression = 'lz4'
 
 
+def _extract_big_bytes(x, big, path=()):
+    if type(x) is dict:
+        for k, v in x.items():
+            if isinstance(v, (list, dict)):
+                _extract_big_bytes(v, big, path + (k,))
+            elif type(v) is bytes and len(v) >= 2**31:
+                big[path + (k,)] = v
+                del x[k]
+    elif type(x) is list:
+        for k, v in enumerate(x):
+            if isinstance(v, (list, dict)):
+                _extract_big_bytes(v, big, path + (k,))
+            elif type(v) is bytes and len(v) >= 2**31:
+                big[path + (k,)] = v
+                x[k] = None
+
+
 def dumps(msg):
```

Comment:
These times are probably somewhat non-trivial.  Current peak performance is around 5000 tasks per second, which gives us a 200 us budget per task. 

Here is a benchmarking script that can be run with snakeviz

https://gist.github.com/5b2b6dcc92d760003f1d9b393f8d4b13

```
python -m cProfile -o prof.out distributed-benchmark.py
snakeviz prof.out
```